__docs__/wonder-blocks-switch/switch.stories.tsx

```diff
@@ -0,0 +1,167 @@
+    const [checkedOne, setCheckedOne] = React.useState(false);
+    const [checkedTwo, setCheckedTwo] = React.useState(false);
+
+    return (
+        <View style={styles.column}>
+            <Switch checked={checkedOne} onChange={setCheckedOne} />
+            <Strut size={Spacing.xSmall_8} />
+            <Switch
+                testId="test-switch"
+                ariaLabel="test switch"
+                checked={checkedTwo}
+                onChange={setCheckedTwo}
+                icon={<Icon icon={icons.search} />}
+            />
+        </View>
+    );
+};
+
+Controlled.play = async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+
+    const switchWithIcon = canvas.getByTestId("test-switch");
+    const switchInput = canvas.getByRole("switch", {name: "test switch"});
+
+    await userEvent.tab();
+    await userEvent.tab();
+
+    expect(switchWithIcon).toHaveStyle(
+        "background-color: rgba(33, 36, 44, 0.5)",
+    );
+    expect(switchWithIcon).toHaveStyle("outline: 2px solid rgb(24, 101, 242)");
+    expect(switchInput).toHaveAttribute("aria-checked", "false");
+
+    await userEvent.click(switchWithIcon);
+
+    expect(switchWithIcon).toHaveStyle("background-color: rgb(24, 101, 242)");
+    expect(switchInput).toHaveAttribute("aria-checked", "true");
+};
+
+Controlled.parameters = {
+    docs: {
+        description: {
+            story: `The switch is a controlled component, so state should be used to keep track of
+            whether it is checked or not. The \`onChange\` prop is optional in case the toggle
+            will be wrapped in a larger clickable component.`,
+        },
+    },
+};
+
+export const Disabled: StoryComponentType = () => (
+    <View style={styles.column}>
+        <Switch checked={false} disabled={true} />
+        <Strut size={Spacing.xSmall_8} />
+        <Switch checked={true} disabled={true} />
+        <Strut size={Spacing.xSmall_8} />
+        <Switch
+            checked={false}
+            disabled={true}
+            icon={<Icon icon={icons.search} />}
+        />
+        <Strut size={Spacing.xSmall_8} />
+        <Switch
+            checked={true}
+            disabled={true}
+            icon={<Icon icon={icons.search} />}
+        />
+    </View>
+);
+
+Disabled.parameters = {
+    docs: {
+        description: {
+            story: `The switch can be disabled.`,
+        },
+    },
+};
+
+export const WithIcon: StoryComponentType = () => {
+    return (
+        <View style={styles.column}>
+            <Switch checked={false} icon={<Icon icon={icons.search} />} />
+            <Strut size={Spacing.xSmall_8} />
+            <Switch checked={true} icon={<Icon icon={icons.search} />} />
+        </View>
+    );
+};
+
+WithIcon.parameters = {
+    docs: {
+        description: {
+            story: `The \`medium\` size switch can take an \`icon\` prop, which will be rendered inside the slider.`,
+        },
+    },
+};
+
```

Comment:
Do you think you could add a `WithLabel` story to demonstrate how this would be used with a `<label>`? Note that this would have to be updated after the `LabeledField` component is created.